tracing/tracing.go

```diff
@@ -183,25 +184,52 @@ func getClient(tracingCfg config.TracingConfig) (otlptrace.Client, error) {
 		if tracingCfg.Insecure {
 			opts = append(opts, otlptracegrpc.WithInsecure())
 		}
+		if tracingCfg.Compression != "" {
+			opts = append(opts, otlptracegrpc.WithCompressor(tracingCfg.Compression))
```

Comment:
Why is this different than http?

Reply:
In the upstream (I'm unsure there is a real reason beyond oversight), the compression options differ between gRPC and HTTP: gRPC option `WithCompressor` takes a string with compression name; HTTP option `WithCompression` takes a typed int `Compression` representing the compression type. 

Both accept only `gzip` (or no compression), it's just the way it is conveyed in the option is different.

Reply:
We should then error if the user passes something else than gzip.

Reply:
I wanted to leave this aligned with how Otel options handle this case, i.e. if invalid option is provided, no compression is used (and an error is logged). But if you think erroring out is more convenient, we can change this behavior.

Reply:
yes, erroring out is how we do things Prometheus-side.

Reply:
Understood :+1: I adjusted the code.